### ISSUE TITLE
Enable RefinedAliasSets on Z and ARM?

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/openj9#6442]
Only x86 and Power enable refined alias sets (via the CodeGenerator flag `setEnableRefinedAliasSets`).  Is this by design, or an oversight?  What does a CodeGenerator need to do to support refined alias sets?

@vijaysun-omr @andrewcraik @fjeremic @gita-omr @knn-k

**Comment by fjeremic** [comment]
Can someone also explain what this optimization is for? If we can get some context on what it's trying to improve I'll be able to tell why/why not this may be needed on Z, or if it was just something we missed implementing.

**Comment by vijaysun-omr** [comment]
I'm not sure the flag does in fact need to reside in the code generator object. If you look at the logic for the feature in question, the "refining" of alias sets happens by walking the final trees in the method. As a result the code to perform this walk might have been done as part of (or close to when) the lowerTrees pass. lowerTrees was a routine in the code generator and so the rationale for putting the query on the code generator may not have been much more complex than that.

**Comment by 0xdaryl** [comment]
@vijaysun-omr, if refined aliases happens late in the optimizer (or even lower trees), who consumes the aliasing information?  Perhaps this is getting more to @fjeremic's question above.

**Comment by vijaysun-omr** [comment]
I believe refining of alias information done at that last stage is persisted past the end of the compilation. i.e. a later compilation that sees a call to a method M whose IL trees were peeked when M was being compiled would have access to the refined aliases. 

For what it's worth we also used to consume aliasing in the instruction scheduler in the days when that pass was still used in the code generators.

**Comment by 0xdaryl** [comment]
I wonder if there is even a need for a Code Generator flag to control this then.  Sounds like the `TR_DisableRefinedAliases` option setting should be all we need to control this.  If some architecture, for some reason, chooses not to support this then it would disable that option flag as we do in so many other places.

**Comment by vijaysun-omr** [comment]
I would agree with that direction since I don't actually believe there is any specific code generation capability that this depends on.
